app/routes/movies.$id.jsx

```diff
@@ -1,19 +1,72 @@
 import { json } from "@remix-run/node";
-import { useLoaderData } from "@remix-run/react";
+import { Link, isRouteErrorResponse, useRouteError, useLoaderData } from "@remix-run/react";
+import styles from "../styles/movie-details.css?url";
 
 export const loader = async ({ params }) => {
-  const res = await fetch(`https://freetestapi.com/api/v1/movies/${params.id}`);
-  const movie = await res.json();
-  return json(movie);
+  try {
+    const res = await fetch(`https://freetestapi.com/api/v1/movies/${params.id}`);
+    const movie = await res.json();
+    return json(movie);
+  } catch(err) {
+    console.log(err);
+    throw new Response(null, {
+      status: 404,
+      statusText: "Not Found",
```

Comment:
Why not found? You can predict what error will happen?

Reply:
You can use this exact message here only if returned status code is 404 otherwise you should take message from the error or set default error text, something like "Something unexpected happened"